Tasks/AzureMonitorAlerts/azurerestutility.ts

```diff
@@ -0,0 +1,265 @@
+    }
+
+    // Add the hidden link resource tag 
+    alertRuleResourceTags["hidden-link:" + resourceUri] = "Resource";
+    
+    // Add email action
+    if(notifyViaEmails) {
+        let notifyEmailsList: string[] = !!notifyEmails ? notifyEmails.split(";") : []; // BUG server throws 400 error, check with team 
+        alertRuleActions.push({ 
+            "odata.type": "Microsoft.Azure.Management.Insights.Models.RuleEmailAction",
+            "sendToServiceOwners": notifyServiceOwners, 
+            "customEmails": notifyEmailsList 
+        });
+    }
+
+    return {
+        location: alertRuleResourceLocation,
+        tags: alertRuleResourceTags,
+        properties: {
+            name: rule.alertName,
+            description: `Updated via ${Util.getDeploymentUri()}`, 
+            isEnabled: true,
+            condition: {
+                "odata.type": "Microsoft.Azure.Management.Insights.Models.ThresholdRuleCondition",
+                "dataSource": {
+                    "odata.type": "Microsoft.Azure.Management.Insights.Models.RuleMetricDataSource",
+                    "resourceUri": resourceUri,
+                    "metricName": rule.metric.value,
+                    "operator": Util.getThresholdConditionForMetricAlertRule(rule.thresholdCondition)
+                },
+                "threshold": rule.thresholdValue,
+                "windowSize": Util.getWindowSizeForMetricAlertRule(rule.timePeriod)
+            },
+            actions: alertRuleActions 
+        }
+    } as IAzureMetricAlertRequestBody;
+}
+
+export async function addOrUpdateAzureMetricAlertRule(
+    endpoint, 
+    resourceGroupName: string, 
+    resourceUri: string, 
+    rule: IAzureMetricAlertRule, 
+    notifyServiceOwners: boolean, 
+    notifyEmails: string) : Promise<any> {
+   
+    let deferred: Q.Deferred<any> = Q.defer<any>();
+    let accessToken: string = await getAuthorizationToken(endpoint);
+   
+    let requestBody: IAzureMetricAlertRequestBody = await getRequestBodyForAddingAlertRule(endpoint, resourceGroupName, resourceUri, rule, notifyServiceOwners, notifyEmails);
+    tl.debug(`Sending PUT request with request body \n${JSON.stringify(requestBody, null, 2)}`); 
+
+    let apiVersion = "2016-03-01";
+    let restUrl = `${endpoint.url}subscriptions/${endpoint.subscriptionId}/resourceGroups/${resourceGroupName}/providers/Microsoft.insights/alertrules/${rule.alertName}?api-version=${apiVersion}`;
+    let restOptions: IRequestOptions = {
+        additionalHeaders: {
+            "Authorization": "Bearer " + accessToken
+        }
+    };
+
+    restObj.replace(restUrl, requestBody, restOptions)
+        .then((response) => {
+            if(response.statusCode === 200 || response.statusCode === 201) {
+                deferred.resolve(response);
+            }
+            else {
+                tl.debug(`Updating the rule ${rule.alertName} failed. ${JSON.stringify(response.result, null, 2)}`);
```

Comment:
log the response object. that will be more informative than just the statusCode

Reply:
Yup will do, thanks.